paasta_tools/kubernetes/application/controller_wrappers.py

```diff
@@ -216,37 +217,31 @@ def create(self, kube_client: KubeClient) -> None:
         self.sync_horizontal_pod_autoscaler(kube_client)
 
     def deep_delete_and_create(self, kube_client: KubeClient) -> None:
-        self.deep_delete(kube_client)
-        timer = 0
-        while (
-            self.kube_deployment in set(list_all_deployments(kube_client))
-            and timer < 60
-        ):
-            sleep(1)
-            timer += 1
-
-        if timer >= 60 and self.kube_deployment in set(
-            list_all_deployments(kube_client)
-        ):
-            try:
-                force_delete_pods(
-                    self.item.metadata.name,
-                    self.kube_deployment.service,
-                    self.kube_deployment.instance,
-                    self.item.metadata.namespace,
-                    kube_client,
-                )
-            except ApiException as e:
-                if e.status == 404:
-                    # Deployment does not exist, nothing to delete but
-                    # we can consider this a success.
-                    self.logging.debug(
-                        "not deleting nonexistent deploy/{} from namespace/{}".format(
-                            self.kube_deployment.service, self.item.metadata.namespace
-                        )
+        # When deleting then immediately creating, we need to use Background deletion to ensure we can create the deployment immediately
+        self.deep_delete(kube_client, propagation_policy="Background")
+        # Also forcibly delete remaining pods
```

Comment:
nit: We are losing the 60s wait on GC pod deletion, but that's okay since this fn is only called for brutal bounces. I would add a comment here, but not that impt.

Reply:
Hm, this is a good point; we may have some services that use brutal but still need to have a <60s grace period. I originally thought this was strictly to let GC do its job, rather than allowing for graceful pod deletions.

However we've seen the problem is that GC may not kick in immediately after attempting the deployment deletion anyway -- should we instead try to delete pods, wait 60s, then force delete pods to be more explicit?